src/libraries/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -6,21 +6,24 @@
 using System.IO;
 using System.Collections;
 using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
 
 namespace System.Diagnostics
 {
     internal static class TraceInternal
     {
         private class TraceProvider : DebugProvider
         {
-            public override void Fail(string message, string detailMessage) { TraceInternal.Fail(message, detailMessage); }
+#pragma warning disable CS8770 // Method lacks `[DoesNotReturn]` annotation to match overridden member.
+            public override void Fail(string? message, string? detailMessage) { TraceInternal.Fail(message, detailMessage); }
```

Comment:
Didn't added `[DoesNotReturn]` because the method called here `TraceInternal.Fail(message, detailMessage);` is not throwing, seems better just suppress warning

Reply:
Wouldn't that maybe suggest that the `[DoesNotReturn]` annotation in the base class is incorrect? Do we cast instances of `TraceProvider` to its base class anywhere in the code?

Reply:
> Wouldn't that maybe suggest that the [DoesNotReturn] annotation in the base class is incorrect? 

Seems `DebugProvider.Fail(...)` eventually throws by calling `DebugProvider.FailCore(...)`, but somehow `DebugProvider.FailCore(...)` is not annotated with `[DoesNotReturn]` instead compiler warning is disabled https://github.com/dotnet/runtime/blob/0f834db1fd80cf82e5ef27f72c48af1c911775da/src/libraries/System.Private.CoreLib/src/System/Diagnostics/DebugProvider.cs#L31

Anyways for this method it is not throw, just log the message to the listeners

> Do we cast instances of TraceProvider to its base class anywhere in the code?

Not look like we do